microsoft/BotFramework-WebChat
# Fix activity grouping for accessibility

> Fixes #3133.

## Changelog Entry

### Fixed

-  Fixes [#3133](https://github.com/microsoft/BotFramework-WebChat/issues/3133). Associate ARIA labels with buttons in hero card and Adaptive Cards, by [@compulim](https://github.com/compulim) in PR [#3146](https://github.com/microsoft/BotFramework-WebChat/issues/3146).
   -  Remove browser-based detection from `<ScreenReaderText>` because it is no longer needed.
   -  After stripping Markdown syntax for accessibility labels, cache the result to improve rendering performance.
   -  Skip stripping Markdown for non-Markdown text content.

## Description

When AT is reading Adaptive Card attachment, the label (`<ScreenReaderText>`) is not read correctly.

On top of this fix, we also need Adaptive Card bump that would remove `tabindex="0'` from their DOM node.

## Specific Changes

Please refer to [changelog](#changelog) above for specific changes.

---

-  [x] ~Testing Added~
   -  Manual testing steps are required
      1. Bot send 2 different hero cards (type `herocard qna` to MockBot)
         - The activity which contain the hero cards must have a text on it
         - The content of the hero card must have 2 card actions
      1. Launch Narrator or other AT
      1. Focus on the first button of first hero card
      1. Press <kbd>TAB</kbd> to focus to the first button of the second hero card
      1. Make sure AT read "Bot said, second activity text. Sent at ... First button"
      1. Press <kbd>TAB</kbd> to focus to the second button of the second hero card
      1. Make sure AT read "Second button"
      1. Press <kbd>SHIFT</kbd> + <kbd>TAB</kbd> to focus to the second button of the first hero card
      1. Make sure AT read "Bot said, first activity text. Sent at ... Second button"
